erpnext/stock/doctype/stock_entry/stock_entry.js

```diff
@@ -86,6 +86,12 @@ frappe.ui.form.on('Stock Entry', {
 		if(frm.doc.company) {
 			frm.trigger("toggle_display_account_head");
 		}
+
+		if(frm.doc.docstatus==1 && frm.doc.purpose == "Material Receipt") {
+			frm.add_custom_button(__('Make Retention Stock Entry'), function () {
```

Comment:
Should "Make Retention Stock Entry" be available in Purchase Receipt as well?

Reply:
Yup makes sense, added it to Purchase Receipt as well.